File: vdp-c-sharp/vdp-c-sharp/VDPApiTest/vta/ReplaceCardTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Configuration;
using System.Collections.Generic;

namespace Vdp
{
    [TestClass]
    public class ReplaceCardTest
    {
        string replaceCardsRequest;
        private VisaAPIClient visaAPIClient;

        public ReplaceCardTest()
        {
            visaAPIClient = new VisaAPIClient();
            string strDate = DateTime.UtcNow.ToString("yyyy-MM-ddThh:mm:ss.fffZ");
            replaceCardsRequest = "{"
                         + "\"communityCode\": \"" + ConfigurationManager.AppSettings["vtaCommunityCode"] + "\","
                                         + "\"newCard\": {"
                                             + "\"address\":" + ConfigurationManager.AppSettings["vtaReplaceCardNewAddress"]  + ","
                                             + "\"billCycleDay\": \"22\","
                                             + "\"bin\": null,"
                                             + "\"cardEnrollmentDate\": \"2016-06-10T08:36:59+00:00\","
                                             + "\"cardExpiryDate\": \"" + ConfigurationManager.AppSettings["vtaReplaceCardExpiryDate"] + "\","
                                             + "\"cardNickName\": \"My Visa 3\","
                                             + "\"cardNumber\": \"" + ConfigurationManager.AppSettings["vtaReplaceCardNumber"] + "\","
                                             + "\"cardSecurityCode\": \"" + ConfigurationManager.AppSettings["vtaReplaceCardSecurityCode"] + "\","
                                             + "\"isActive\": true,"
                                             + "\"lastFour\": \"" + ConfigurationManager.AppSettings["vtaReplaceCardLastFour"] + "\","
                                             + "\"nameOnCard\": \"Mradul\","
                                             + "\"paused\": false,"
                                             + "\"portfolioNum\": \"" + ConfigurationManager.AppSettings["vtaPortfolioNumber"] + "\","
                                             + "\"previousCardNumber\": null,"
                                             + "\"productId\": null,"
                                             + "\"productIdDescription\": \"Credit\","
                                             + "\"productType\": \"Credit\","
                                             + "\"productTypeExtendedCode\": \"123\","
                                             + "\"rpin\": null"
                                         + "},"
                                         + "\"oldCard\": {"
                                             + "\"address\":" + ConfigurationManager.AppSettings["vtaReplaceCardNewAddress"] + ","
                                             + "\"billCycleDay\": \"22\","
                                             + "\"bin\": null,"
                                             + "\"cardEnrollmentDate\": \"2016-06-10T08:36:59+00:00\","
                                             + "\"cardExpiryDate\": \"" + ConfigurationManager.AppSettings["vtaCreateCustomerCardExpiryDate"] + "\","
                                             + "\"cardNickName\": \"My Visa 3\","
                                             + "\"cardNumber\": \"" + ConfigurationManager.AppSettings["vtaCreateCustomerCardNumber"] + "\","
                                             + "\"cardSecurityCode\": \"" + ConfigurationManager.AppSettings["vtaCreateCustomerCardSecurityCode"] + "\","
                                         + "\"isActive\": true,"
                                         + "\"lastFour\": \"" + ConfigurationManager.AppSettings["vtaCreateCustomerLastFour"] + "\","
                                         + "\"nameOnCard\": \"ddd\","
                                         + "\"paused\": false,"
                                         + "\"portfolioNum\": \"" + ConfigurationManager.AppSettings["vtaPortfolioNumber"] + "\","
                                         + "\"previousCardNumber\": null,"
                                         + "\"productId\": null,"
                                         + "\"productIdDescription\": \"Credit\","
                                         + "\"productType\": \"Credit\","
                                         + "\"productTypeExtendedCode\": \"123\","
                                         + "\"rpin\": null"
                                       + "}"
                             + "}";
        }

        [TestMethod]
        public void TestReplaceCard()
        {
            string baseUri = "vta/";
            string resourcePath = "v3/communities/" + ConfigurationManager.AppSettings["vtaCommunityCode"] + "/cards";
            Dictionary<string, string> headers = new Dictionary<string, string>();
            headers.Add("ServiceId", ConfigurationManager.AppSettings["vtaServiceId"]);
            string status = visaAPIClient.DoMutualAuthCall(baseUri + resourcePath, "POST", "Replace a card test", replaceCardsRequest, headers);
            Assert.AreEqual(status, "Created");
        }
    }
}

File: vdp-node/test/vta/replacecardtest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Replace Cards', function() {
	var visaAPIClient = new VisaAPIClient();
	var date = new Date().toISOString();
	var replaceCardsRequest = JSON.stringify({
        "communityCode": config.vtaCommunityCode,
        "newCard": {
            "address": config.vtaReplaceCardNewAddress,
            "billCycleDay": "22",
            "bin": null,
            "cardEnrollmentDate": "2016-06-10T08:36:59+00:00",
            "cardExpiryDate": config.vtaReplaceCardExpiryDate,
            "cardNickName": "My Visa 3",
            "cardNumber": config.vtaReplaceCardNumber,
            "cardSecurityCode": config.vtaReplaceCardSecurityCode,
            "isActive": true,
            "lastFour": config.vtaReplaceCardLastFour,
            "nameOnCard": "Mradul",
            "paused": false,
            "portfolioNum": config.vtaPortfolioNumber,
            "previousCardNumber": null,
            "productId": null,
            "productIdDescription": "Credit",
            "productType": "Credit",
            "productTypeExtendedCode": "123",
            "rpin": null
        },
        "oldCard": {
            "address": config.vtaCreateCustomerAddress,
            "billCycleDay": "22",
            "bin": null,
            "cardEnrollmentDate": "2016-06-10T08:36:59+00:00",
            "cardExpiryDate": config.vtaCreateCustomerCardExpiryDate,
            "cardNickName": "My Visa 3",
            "cardNumber": config.vtaCreateCustomerCardNumber,
            "cardSecurityCode": config.vtaCreateCustomerCardSecurityCode,
        "isActive": true,
        "lastFour":  config.vtaCreateCustomerLastFour,
        "nameOnCard": "ddd",
        "paused": false,
        "portfolioNum": config.vtaPortfolioNumber,
        "previousCardNumber": null,
        "productId": null,
        "productIdDescription": "Credit",
        "productType": "Credit",
        "productTypeExtendedCode": "123",
        "rpin": null
      }
});

	it('Replace Cards Test',function(done) {
		this.timeout(10000);
		var baseUri = 'vta/';
		var resourcePath = 'v3/communities/'+ config.vtaCommunityCode +'/cards';
		visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, replaceCardsRequest, 'POST', {'ServiceId' : config.vtaServiceId}, 
		function(err, responseCode) {
			if(!err) {
				assert.equal(responseCode, 200);
			} else {
				assert(false);
			}
			done();
		});
	});
});
